tests/integration/zenodo_test.py

```diff
@@ -143,9 +147,43 @@ async def test_zenodo_workflow(
     mocker,
 ):
     """Test the entire zenodo client workflow."""
-    from pudl_archiver.zenodo.api_client import ZenodoClient
 
-    client = ZenodoClient(test_settings, session, upload_key, publish_key, testing=True)
+    def verify_files(expected, deposition: Deposition):
+        deposition_files = {f.filename: f for f in deposition.files}
+        for file_data in expected:
+            # Verify that all expected files are in deposition
+            assert file_data["path"].name in deposition_files
+
+            # Download each file
+            file_link = deposition_files[file_data["path"].name].links.download
+            res = requests.get(file_link, params={"access_token": upload_key})
+
+            # Verify that contents of file are correct
+            assert res.text.encode() == file_data["contents"]
+
+    deposition_interface_args = {
+        "data_source_id": "pudl_test",
+        "session": session,
+        "upload_key": upload_key,
+        "publish_key": publish_key,
+        "deposition_settings": test_settings,
+        "dry_run": False,
+        "sandbox": True,
+    }
+
+    class TestDownloader(AbstractDatasetArchiver):
+        name = "Test Downloader"
+
+        def __init__(self, resources, **kwargs):
+            super().__init__(**kwargs)
+            self.resources = resources
+
+        async def get_resources(self):
+            async def identity(x):
```

Comment:
This seems like an awkward way to `async`-ify values, but I haven't figured out anything nicer.

Reply:
There might be a cleaner way, but I don't really feel like it's worth the effort for a test haha